crates/polars-io/src/file_cache/cache_lock.rs

```diff
@@ -0,0 +1,173 @@
+use std::path::PathBuf;
+use std::sync::atomic::AtomicBool;
+use std::sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard};
+use std::time::Duration;
+
+use fs4::FileExt;
+use once_cell::sync::Lazy;
+use polars_core::config;
+
+use crate::pl_async;
+
+pub(super) static GLOBAL_FILE_CACHE_LOCK: Lazy<GlobalLock> = Lazy::new(|| {
+    let path = std::env::var("POLARS_TEMP_DIR")
+        .unwrap_or_else(|_| std::env::temp_dir().to_string_lossy().into_owned());
+    let path = PathBuf::from(path).join("polars/file-cache/");
+    let _ = std::fs::create_dir_all(&path);
+    let path = path.join(".process-lock");
+
+    let file = std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(false)
+        .open(path)
+        .map_err(|err| {
+            panic!("failed to open/create global file cache lockfile: {}", err);
+        })
+        .unwrap();
+
+    let at_bool = Arc::new(AtomicBool::new(false));
+    // Holding this access tracker prevents the background task from
+    // unlocking the lock.
+    let access_tracker = AccessTracker(at_bool.clone());
+
+    pl_async::get_runtime().spawn(async move {
+        let access_tracker = at_bool;
+        let verbose = config::verbose();
+
+        loop {
+            if !access_tracker.swap(false, std::sync::atomic::Ordering::Relaxed)
+                && GLOBAL_FILE_CACHE_LOCK.try_unlock()
+                && verbose
+            {
+                eprintln!("unlocked global file cache lockfile");
+            }
```

Comment:
Before sleeping, I think we can also set a notify: https://docs.rs/tokio/latest/tokio/sync/struct.Notify.htm;

As I understand it this will not schedule this task until we notify that we have set the locked file first. If that works I think we can set the `sleep` shorter to 200 ms or something like that.

Reply:
Thanks for the tip - I've added the notify layer for the background task to await on, which should make it more efficient.